blocks/gallery-block/features/gallery/default.jsx

```diff
@@ -1,35 +1,65 @@
-import React from 'react';
+import React, { lazy, Suspense } from 'react';
 import PropTypes from 'prop-types';
-import { useFusionContext } from 'fusion:context';
+import { useContent } from 'fusion:content';
+import { useFusionContext, useAppContext } from 'fusion:context';
 import getProperties from 'fusion:properties';
 import getTranslatedPhrases from 'fusion:intl';
-import GlobalContentGallery from './_children/global-content';
-import CustomContentGallery from './_children/custom-content';
+
+import { Gallery, ErrorBoundary } from '@wpmedia/engine-theme-sdk';
+
+const AdFeature = lazy(/* istanbul ignore next */ () => import('@wpmedia/ads-block'));
 
 const GalleryFeature = (
   {
     customFields: {
       inheritGlobalContent,
       galleryContentConfig,
     } = {},
-  } = {},
+  },
 ) => {
   const { arcSite } = useFusionContext();
-  const { locale = 'en' } = getProperties(arcSite);
+  const { resizerURL, galleryCubeClicks, locale = 'en' } = getProperties(arcSite);
+  const { globalContent = {} } = useAppContext();
   const phrases = getTranslatedPhrases(locale);
+  const content = useContent(galleryContentConfig ? {
+    source: galleryContentConfig.contentService,
+    query: galleryContentConfig.contentConfigValues,
+  } : {});
 
-  let showGlobalContent;
-  if (typeof inheritGlobalContent === 'undefined') {
-    showGlobalContent = (typeof galleryContentConfig === 'undefined');
-  } else {
-    showGlobalContent = inheritGlobalContent;
-  }
+  const showGlobalContent = (typeof inheritGlobalContent === 'undefined') ? (typeof galleryContentConfig === 'undefined') : inheritGlobalContent;
+  const {
+    content_elements: contentElements = [],
+    headlines = {},
+    id = '',
+  } = showGlobalContent ? globalContent : content;
 
-  if (showGlobalContent) {
-    return <GlobalContentGallery phrases={phrases} />;
-  }
+  const interstitialClicks = parseInt(galleryCubeClicks, 10);
 
-  return <CustomContentGallery contentConfig={galleryContentConfig} phrases={phrases} />;
+  return (
+    <Gallery
+      galleryElements={contentElements}
+      resizerURL={resizerURL}
+      ansId={id}
+      ansHeadline={headlines?.basic ? headlines.basic : ''}
+      expandPhrase={phrases.t('global.gallery-expand-button')}
+      autoplayPhrase={phrases.t('global.gallery-autoplay-button')}
+      pausePhrase={phrases.t('global.gallery-pause-autoplay-button')}
+      pageCountPhrase={/* istanbul ignore next */ (current, total) => phrases.t('global.gallery-page-count-text', { current, total })}
+      adElement={/* istanbul ignore next */ () => (
+        <ErrorBoundary fallback={<div>Missing Ad block</div>}>
+          <Suspense fallback={<div>Missing ad block</div>}>
```

Comment:
Very small nitpick but I would expect to see a loading state while this is about to be successfully loaded. I guess `fallback` is a bit of a misnomer for suspense 

```suggestion
          <Suspense fallback={<div>Loading ad block</div>}>
```